### ISSUE TITLE
Story variables no longer set correctly

### ISSUE THREAD

**Issue by rabidgremlin** [bladecoder/blade-ink#15]
Versions of blade-ink after 0.4.4 no longer set variables correctly.

I have the following ink script:

```
VAR x = ""

This is a test
SET_X:
{
  - x == "":
    -> x_not_set
  - else:
    -> x_is_set
}
-> END

= x_not_set
X is not set!
-> END

= x_is_set
X is set
-> END
```

With blade-ink:0.4.3 and blade-ink:0.4.4 the output is:
```
This is a test
X is set
```
with the following "driver code":
```
 Story story = new Story(sourceJsonString);
		
    while (story.canContinue()) 
    {
        String line = story.Continue();

        if (line.startsWith("SET_X:"))
        {
            story.getVariablesState().set("x", 100);
        }
        else
        {
            System.out.print(line);
        }            
    }
```

With blade-ink:0.5.0 and  blade-ink:0.5.1, the output is:
```
This is a test
X is not set!
```

This appears to be a blade-ink issue as using latest Ink version (~~0.8.2~~ 0.8.3) in Unity produces the correct output of:
```
This is a test
X is set
```

With the following C# driver code:
```
void Awake()
    {
        _inkStory = new Story(inkAsset.text);

        while (_inkStory.canContinue)
        {
            string line = _inkStory.Continue();

            if (line.StartsWith("SET_X:"))
            {
              _inkStory.variablesState["x"] = 100;
            }
            else
            {
              Debug.Log(line);
            }
        }
    }
```

**Comment by rabidgremlin** [comment]
One other quirk is that if the Ink script is modified to output the value of the variable prior to the condition then it work with the latest blade-ink version:
```
VAR x = ""

This is a test
SET_X:
X is {x}
{
  - x == "":
    -> x_not_set
  - else:
    -> x_is_set
}
-> END

= x_not_set
X is not set!
-> END

= x_is_set
X is set
-> END
```

Give output of:
```
This is a test
X is 100
X is set
```

**Comment by rabidgremlin** [comment]
Test code: [TestBladeInk.zip](https://github.com/bladecoder/blade-ink/files/3239097/TestBladeInk.zip)

**Comment by bladecoder** [comment]
Thanks for your reporting! This is a weird bug, I will look at it as soon as possible.

**Comment by bladecoder** [comment]
I test this bug with the v0.8.3 inklecate:

```
$ ~/programs/ink/inklecate.exe -p src/test/resources/inkfiles/misc/issue15.ink
This is a test
SET_X:
X is not set!
```

So the error is not in the blade-ink, it is in the Inkle reference implementation. We have to open the issue in the official ink repository: https://github.com/inkle/ink/

**Comment by bladecoder** [comment]
Could you test in Unity with the v0.8.3 to see if this is a problem with this version?

**Comment by rabidgremlin** [comment]
It worked with Unity and v0.8.2. Will give it a try with 0.8.3 in Unity

**Comment by rabidgremlin** [comment]
Also tried out the older v0.8.1 Unity package and it also works.

**Comment by bladecoder** [comment]
I found the error! Fixed in v0.5.2

Thanks!
